prbt_hardware_support/src/libmodbus_client.cpp

```diff
@@ -35,15 +36,45 @@ LibModbusClient::~LibModbusClient()
 
 bool LibModbusClient::init(const char* ip, unsigned int port)
 {
+  long   arg;
+  int    sockfd;
+  int    retries;
+  struct sockaddr_in serv_addr;
+  
+  sockfd = socket(AF_INET, SOCK_STREAM, 0);      // Create socket for testing purpose
+  bzero((char *) &serv_addr, sizeof(serv_addr)); 
+  serv_addr.sin_family = AF_INET;                   
+  serv_addr.sin_port = htons((short unsigned int)port);
+  
+  arg = fcntl(sockfd, F_GETFL, NULL); 
+  arg |= O_NONBLOCK; 
+  fcntl(sockfd, F_SETFL, arg);                   //set connection to non blocking, no timeout
```

Comment:
Maybe better rename the auxiliary variable arg to result

```suggestion
  long   result; // auxiliary variable
  int    sockfd;
  int    retries;
  struct sockaddr_in serv_addr;
  
  sockfd = socket(AF_INET, SOCK_STREAM, 0);      // Create socket for testing purpose
  bzero((char *) &serv_addr, sizeof(serv_addr)); 
  serv_addr.sin_family = AF_INET;                   
  serv_addr.sin_port = htons((short unsigned int)port);
  
  result = fcntl(sockfd, F_GETFL, NULL); 
  result |= O_NONBLOCK; 
  fcntl(sockfd, F_SETFL, result);                   //set connection to non blocking, no timeout
```